### ISSUE TITLE
[Inductor][CSE] `split_with_sizes` not deduplicated on B200 with FP8 quantization

### ISSUE THREAD

**Issue by ikchifo** [pytorch/pytorch#174472]
### Describe the bug

**Summary:**
When compiling models with FP8 quantization on NVIDIA B200 GPUs, Inductor's CSE pass fails to deduplicate identical `split_with_sizes` calls. The graph contains 3 separate split nodes with 1 user each, instead of 1 split node with 3 users.

**Context:**
- Affects: CSE pass in Inductor
- GPU: NVIDIA B200 (Blackwell)
- Dtype: FP8 (via vLLM's quantization path)
- Impact: Breaks downstream pattern matching that expects deduplicated nodes

### Minimal Reproducible Example

```python
import torch

# Simulates the pattern seen in vLLM's QK norm + RoPE fusion
# On B200+FP8, the three split_with_sizes calls are NOT deduplicated by CSE

def model(qkv):
    # These should be CSE'd into a single split node
    split1 = torch.ops.aten.split_with_sizes.default(qkv, [512, 128, 128], dim=-1)
    split2 = torch.ops.aten.split_with_sizes.default(qkv, [512, 128, 128], dim=-1)
    split3 = torch.ops.aten.split_with_sizes.default(qkv, [512, 128, 128], dim=-1)
    
    q = split1[0]  # First user
    k = split2[1]  # Second user  
    v = split3[2]  # Third user
    return q, k, v

compiled = torch.compile(model, backend="inductor")
qkv = torch.randn(1024, 768, dtype=torch.bfloat16, device="cuda")
compiled(qkv)
```

*Note: Actual repro requires B200 hardware with FP8 model. The issue was observed in vLLM with `Qwen/Qwen3-30B-A3B-FP8`.*

### Expected Behavior

CSE should recognize the three `split_with_sizes` calls are identical (same input tensor, same split sizes, same dim) and deduplicate them into a single node with 3 `getitem` users.

### Actual Behavior

Three separate `split_with_sizes` nodes remain in the FX graph:

```
split_with_sizes_1 = torch.ops.aten.split_with_sizes.default(qkv, [512, 128, 128], -1)
getitem_0 = split_with_sizes_1[0]  # q

split_with_sizes_2 = torch.ops.aten.split_with_sizes.default(qkv, [512, 128, 128], -1)
getitem_1 = split_with_sizes_2[1]  # k

split_with_sizes_3 = torch.ops.aten.split_with_sizes.default(qkv, [512, 128, 128], -1)
getitem_2 = split_with_sizes_3[2]  # v
```

This breaks pattern matchers that expect `_users=3` on a single split node.

### Ablation Testing

- [x] Works correctly on: H100 with BF16/FP16
- [ ] Fails on: B200 with FP8 quantization
- [ ] Likely related to FP8 tensor handling or Blackwell-specific IR generation

### Versions

```
PyTorch: 2.9.1+cu129
GPU: NVIDIA B200
CUDA: 12.9
Python: 3.12
```

### Additional Context

**Workaround:** vLLM implemented a pre-pass to manually coalesce duplicate splits before pattern matching: https://github.com/vllm-project/vllm/pull/33967

**Related vLLM issue:** https://github.com/vllm-project/vllm/issues/33295

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo @jansel @eellison @zou3519 @BoyuanFeng @angelayi
